### PR TITLE
ARROW-2548: Clarify `List<Char>` Array example

### PR DESCRIPTION
**Pull request by fwessels**
`joemark` only spans bytes 0 through to 6, so starting from byte position 7 (not 8) the contents can be unspecified.

(or is byte 7 nulled out or something?)

### CODE REVIEW

**Review by xhochy — APPROVED** [review]
+1, this seems correct but in the case of the `Layout.md` document I always prefer to have a second review.

**Review by pitrou — APPROVED** [review]
LGTM too.

**Comment by wesm** [comment]
@fwessels could you create a JIRA account and let us know your user ID so we can assign this issue to you? Thank you

**Comment by fwessels** [comment]
Hi @wesm that will be frank.wessels 
Kind regards,
-Frank

**Pull request closed without merging**
